cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/CountTest.scala

```diff
@@ -0,0 +1,163 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.docgen
+
+import org.neo4j.cypher.docgen.tooling.DocBuilder
+import org.neo4j.cypher.docgen.tooling.DocumentingTest
+import org.neo4j.cypher.docgen.tooling.ResultAssertions
+
+class CountTest extends DocumentingTest {
+  override def outputPath = "target/docs/dev/ql/"
+
+  override def doc = new DocBuilder {
+    doc("COUNT", "query-count")
+    initQueries("""CREATE (andy:Swedish:Person {name: 'Andy', age: 36, belt: 'white'}),
+                  #(timothy:Person {name: 'Timothy', age: 25, address: 'Sweden/Malmo'}),
+                  #(peter:Person {name: 'Peter', age: 35, email: 'peter_n@example.com'}),
+                  #(andy)-[:KNOWS {since: 2012}]->(timothy),
+                  #(andy)-[:KNOWS {since: 1999}]->(peter),
+                  #(andy)-[:HAS_DOG {since: 2016}]->(:Dog {name:'Andy'}),
+                  #(fido:Dog {name:'Fido'})<-[:HAS_DOG {since: 2010}]-(peter)-[:HAS_DOG {since: 2018}]->(:Dog {name:'Ozzy'}),
+                  #(fido)-[:HAS_TOY]->(:Toy{name:'Banana'})""".stripMargin('#'))
+    synopsis("An count subquery `COUNT` can be used to find out how many times a specified pattern appears in the data.")
+    p(
+      """
+        |* <<count-subqueries, Using count subqueries in WHERE>>
+        |** <<count-subquery-simple-case, Simple count subquery>>
+        |** <<count-subquery-with-where, Count subquery with `WHERE` clause>>
+        |** <<count-subquery-nesting, Nesting count subqueries>>
+        |* <<count-subqueries-other-clauses, Using count subqueries inside other clauses>>
+        |** <<count-subqueries-with-return, Using count subqueries with `RETURN` clause>>
+        |** <<count-subqueries-with-set, Using count subqueries with `SET` clause>>
+        |** <<count-subqueries-with-case, Using count subqueries with `CASE` expression>>
+        |** <<count-subqueries-with-with, Using count subqueries with `WITH` expression>>
+      """.stripMargin)
+    p("""#It can be used in the same way as a path pattern but it allows you to use `WHERE` clauses internally.
+        #A subquery has a scope, as indicated by the opening and closing braces, `{` and `}`.
+        #Any variable that is defined in the outside scope can be referenced inside the subquery's own scope.
+        #Variables introduced inside the subquery are not part of the outside scope and therefore can't be accessed on the outside.""".stripMargin('#'))
+    functionWithCypherStyleFormatting("""COUNT {
+                                        #  [Pattern]
+                                        #  WHERE [Expression]
+                                        #}""".stripMargin('#'))
+    p("The `WHERE` clause is optional.")
+    p("It is worth noting that the `MATCH` keyword can be omitted in subqueries and that the `WHERE` clause is optional.")
```

Comment:
Good point, we should probably align them and removing it from `EXISTS` would require a deprecation. As it is just syntactic sugar, I think it only require works on the surface side, but it might still be worth asking Satia if he has any input on it